### ISSUE TITLE
Reproduce results on machine with CUDA 10

### ISSUE THREAD

**Issue by Tongdongq** [NVIDIA/AMGX#60]
My examples/amgx_capi prints
```
AMGX version 2.0.0.130-opensource
Built on May 17 2019, 14:26:12
Compiled with CUDA Runtime 9.2, using CUDA driver 10.1
```
The NVIDIA docs say CUDA driver means the highest supported version, and CUDA Runtime the actual used version.

The problem is now that I cannot reproduce my previous results, when CUDA 10.1 was not yet installed. For example, using CG on bodyy6.mtx with a custom B vector now requires 843 iterations, versus 568 earlier. cfd2.mtx does not converge at all, where it only took 22 iterations before. Does anyone else have these problems?
If you'd like to run the experiments as well, unpack the zip in examples/ (replacing amgx_capi.c), make and run with
```
gcc -O2 -std=c99 amgx_capi.c -c -I/usr/local/cuda-9.2/include -I../base/include
g++ -O2 amgx_capi.o -o amgx_capi -L/usr/local/cuda-9.2/lib64 -L../build -ldl -L../lib -lamgxsh -Wl,-rpath=../build
./amgx_capi -m bodyy6.mtx -b B.ltx -x X.ltx -c CG.json
./amgx_capi -m cfd2.mtx -c CG.json
```

[debug.zip](https://github.com/NVIDIA/AMGX/files/3196876/debug.zip)

**Comment by marsaev** [comment]
> The NVIDIA docs say CUDA driver means the highest supported version, and CUDA Runtime the actual used version.

That is correct.

> The problem is now that I cannot reproduce my previous results, when CUDA 10.1 was not yet installed. 

You mean when you updated drivers you was not able to reproduce previous results?
1. Did you see similar behaviour with other CUDA programs?
2. Are results of your run deterministic? That means that result (i.e. convergence rate, number of required iterations) is the same from run to run. Note that not every solver is deterministic by nature, but you can add global flag "determinism_flag=1" to your config to force determinism

**Comment by marsaev** [comment]
I tried running your example and result matches with CUDA 9.2 and CUDA 10.1

**Comment by Tongdongq** [comment]
Actually, our system admin updated the drivers. But yes, the current results do not match the results I got 2 months ago (AMGX did not change during that time).
For these experiments, the determinism_flag was 1.
I did not see similar behaviour with other CUDA programs, I have another solver, which is not deterministic, but still produces very similar results.
What results did you get? 843 or 568 iterations for bodyy6? And did cfd2 converge?
